modules/andes-core/broker/src/main/java/org/wso2/andes/subscription/LocalSubscription.java

```diff
@@ -98,9 +98,9 @@ public LocalSubscription(OutboundSubscription subscription, String subscriptionI
         // In case of mock subscriptions, this becomes null
         if (null != subscription) {
             //We need to keep the protocolType in basic subscription for notification
-            if(subscription instanceof AMQPLocalSubscription) {
+            if((AMQPUtils.PROTOCOL_TYPE).equals(subscription.getProtocolType())) {
```

Comment:
1. We're doing enum vs string equals here. This is wrong.
2. Looking at the logic here, this could be simplified to one line 'setProtocolType(subscription.getProtocolType())'